contracts/OlasMech.sol

```diff
@@ -32,14 +32,11 @@ abstract contract OlasMech is Mech, IErrorsMech, ImmutableStorage {
     // Number of total deliveries by this mech
     uint256 public numTotalDeliveries;
     // Reentrancy lock
-    uint256 internal _locked = 1;
+    bool transient locked;
 
-    // Map of undelivered requests counts for corresponding addresses in this agent mech
-    mapping(address => uint256) public mapUndeliveredRequestsCounts;
```

Comment:
I believe this is not needed at all. Undelivered requests for specific requesters can be found in marketplace records / events.